core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1197,7 +1197,8 @@ private boolean checkConsistencyInternal(LockedInodePath inodePath) throws Inval
         return ufs.isDirectory(ufsPath);
       } else {
         return ufs.isFile(ufsPath)
-            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength();
+            && ufs.getFileStatus(ufsPath).getContentLength() == inode.asFile().getLength()
+            && (!ufs.getFingerprint(ufsPath).equals(Constants.INVALID_UFS_FINGERPRINT));
```

Comment:
we could actually check if the fingerprint matches between the ufs and the inode, using `inode.getUfsFingerprint()`.

Reply:
Thanks @yuzhu